### DISCUSSION TITLE
Unable to delete payment method

### DISCUSSION THREAD

**Discussion by NikitaMikhanchuk** [community/community#51933] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hello there. I was trying to delete an added payment method, but it is impossible.

What have I done:

1. Install Copilot
2. Add payment method 
3. Start the Copilot trial period

Then I uninstalled Copilot, cancel the trial period, and now trying to delete an added payment method.

The [tutorial](https://docs.github.com/en/billing/managing-your-github-billing-settings/adding-or-editing-a-payment-method) says I need to tap "Update Payment Method" but I can't find this button
![Screenshot 2023-04-04 at 11 37 11](https://user-images.githubusercontent.com/9729120/229736906-5271696a-d4bb-4e5d-8fab-eb643abb000e.png)

I selected Billings and plans --> Payment method and tap Edit button on Payment Method section, but "Remove" button is absent
![Screenshot 2023-04-03 at 11 14 46](https://user-images.githubusercontent.com/9729120/229738378-a1870af1-340f-452a-bbe1-2d11f4f00274.png)

Could you help me with this problem? Thanks.

**Comment by NJacksonDev** [comment, marked as answer]
That is strange - I would contact GitHub support and let them know about the issue. I'm sure they will help resolve.

https://support.github.com/request

**Reply by NikitaMikhanchuk** [reply]
alright, thank you

**Reply by martintaleski** [reply]
Yep, I wrote to support and they will removed the payment method. Just mention GDPR and data privacy. 

I guess eventually they will get tired, and reintroduce the remove button.

**Reply by Andy024** [reply]
I want to remove the payment method too. can you help me please. 
I feel unsafe because github charge the card automatically after 90 days.

**Comment by NikitaMikhanchuk** [comment]
The reason is that system needs some time to display the remove button for added payment method.

**Comment by TriggerDingus** [comment]
Github does not offer a means to delete payment information. There was another thread on this topic but it has apparently been removed.
Contacting Github support didn't resolve it for me. 

This is a privacy and security issue for users and so if someone wishes to pursue removing this personally identifiable information from their account, I suggest a couple avenues to purse since Github isn't offering a remedy:

If you are in a state with user data privacy laws, submit a complaint here:
https://yourdigitalrights.org/

If your state doesn't have user data privacy laws, submit a complaint through your state's consumer privacy department. You should find a link on your state government's web site.

**Comment by GocherDharma** [comment]
This is very BAD from Github. Does not show a button to Delete the payment info a tactic to keep people attached to github? But whatever it is this is very bad and against user privacy laws.

**Comment by asterism612** [comment]
Same Issue....

**Comment by sakurawald** [comment]
Alright, still same issue.

**Comment by jtown81** [comment]
add paypal as a payment method, then the cc info seems to disappear.  then go and revoke permissions on paypal

**Reply by hedges333** [reply]
wfm... the payment method doesn't go away, but I know they can't charge it

**Comment by rukkus333** [comment]
Same Issue. I highly recommend not using this service. Request complete deletion and cancel your card. This is abusive behavior.

**Comment by ctsstc** [comment]
Fantastic the support links to this which doesn't actually resolve the issue..

**Comment by themhz** [comment]
Same issue here.. github grabed money from my wallet without asking.. how do I cancel the payment?

**Comment by sobczuk-m** [comment]
same here

**Comment by MrEbrahimXD** [comment]
Again same here

**Comment by andostronaut** [comment]
Still same issue here

**Reply by MrEbrahimXD** [reply]
Create a ticket, It actually worked and they replied after 3 hrs. And payment options were deleted

**Comment by TANYT-ORG** [comment]
This issue is a total disgrace to GitHub...

**Comment by deep-space-explorer** [comment]
same here

**Reply by yaqbx** [reply]
Same here. And the worst is that M$ charged me for copilot after trial has ended and i didn't used it.

**Comment by 813gan** [comment]
amazing

**Comment by amrahmed242** [comment]
~600 days, still the same issue! 🙂

**Comment by DanieleTrebe995** [comment]
same issue

**Comment by arkpublic** [comment]
Same issue, this is a BAD practice

**Comment by PatrickRorth** [comment]
Same issue, and I'm on free sub. but are still getting billed 5$ a month
![image](https://github.com/user-attachments/assets/fccbb8e4-c688-4a9d-a8f6-38f6d3460935)

![image](https://github.com/user-attachments/assets/7c6c6d1e-6fe2-4369-9c15-7a0b482a3ed9)

**Comment by LaptevPavelA** [comment]
same issue

**Comment by meat-ball** [comment]
still a problem 7/8/25. it's so skeevy and unsafe to leave your PII/financial info without being able to remove/edit.

**Comment by StarterX4** [comment]
Asked the support for credit card removal, and they removed it in a few minutes without a problem.

**Comment by 1stFalloutBoy** [comment]
I have just opened a ticket to have my credit card information removed I assumed that since I was making a one off donation it wasn't going to store my card info but it has and I am pissed. Github is owned by Microsoft now apparently and if they don't remove my card details as requested I will go both routes Github and Microsoft 24/7 complaining until they do. If they want to store your credit card details they should tell you and get your explicit permission and there should be a way to delete it as well, I believe not having that is illegal. No wonder they didn't want people using Paypal - theft of information is what this is.

**Comment by ruwanthisubs-beep** [comment]
2025 October 27th, Still this issue is there. This is a data protection and consumer rights violation from Microsoft side...

[a comment by an automated account is not shown]
